src/operator-run.c

```diff
@@ -476,6 +481,7 @@ void xnn_compute_gemm(
       cm_stride,
       context->cn_stride,
```

Comment:
Thanks for looking at this. I've checked and I'm pretty sure they're right. cm_stride is the output row stride and cn_stride is the col stride. I tried swapping them and I get the wrong answer.  

Kleidi expects cn_stride to be sizeof(float) as ignores it whereas XNNPack expects it to be the number of columns written by the microkernel. IMO cn_stride should just go since it is meaningless and always set to be nr * sizeof(T)

Reply:
Hi @alankelly 

I recommend trying this:

- Changing `context->k_scaled` to `context->k_scaled / sizeof(float)` . Here you are scaling K by the size of the element. This is not necessary because the KleidiAI micro-kernel accepts K, so try with the original K value.
- Switch the position of the `cm_stride` and `cn_stride` arguments. As you rightly point out, KleidiAI always expects `dst_stride_col` to be equal to `sizeof(float)` but it's not used inside the KleidiAI micro-kernel so we can ignore it. But you do need to change `dst_stride_row` to be equal to `cn_stride` which as you say is `nr * sizeof(float)`.

I've run on a few shapes and it's passing for me. Let me know how you get on.

Reply:
This works! Thanks! I will clean up the code and update the PR today.